File: dart/creational/abstract_factory/20210120/bin/concreteFactoryN.dart
import './abstractFactory.dart';
import 'productGradeA.dart';
import 'productGradeZ.dart';
import 'concreteProductAn.dart';
import 'concreteProductZn.dart';

class ConcreteFactoryN implements AbstractFactory {
  @override
  ProductGradeA createProductFamilyA() {
    return ConcreteProductAn();
  }

  @override
  ProductGradeZ createProductFamilyZ() {
    return ConcreteProductZn();
  }
}

File: typescript/creational/abstract_factory/20210120/concreteFactoryN.ts
import {AbstractFactory} from "./abstractFactory";
import {ProductGradeA} from "./productGradeA";
import {ProductGradeZ} from "./productGradeZ";
import {ConcreteProductAn} from "./concreteProductAn";
import {ConcreteProductZn} from "./concreteProductZn";

export class ConcreteFactoryN implements AbstractFactory {
  createFamilyA(): ProductGradeA {
    return new ConcreteProductAn()
  }

  createFamilyZ(): ProductGradeZ {
    return new ConcreteProductZn()
  }

}
